system/ymodem/ymodem.h

```diff
@@ -0,0 +1,83 @@
+/****************************************************************************
+ * apps/system/ymodem/ymodem.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __APPS_SYSTEM_YMODEM_YMODEM_H
+#define __APPS_SYSTEM_YMODEM_YMODEM_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdint.h>
+#include <stdlib.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifdef CONFIG_SYSTEM_YMODEM_DEBUGFILE_PATH
+#  define ymodem_debug(...) dprintf(ctx->debug_fd, ##__VA_ARGS__)
```

Comment:
However, it is usually transmitted by/dev/console. It is more convenient for using writing files

Reply:
too many logs will be generated during the transmission process. If you continue to write /tmp, it will  consume to much of memory, which is not friendly on devices with limited resources.

Reply:
you can set CONFIG_SYSTEM_YMODEM_DEBUGFILE_PATH to "/dev/console"